diff --git a/BUILD.bazel b/BUILD.bazel
index 15e97d3..277f6ac 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -80,12 +80,12 @@ cc_library(
         "@boost.variant2",
     ] + select({
         ":boringssl": [
-            "@boringssl//:crypto",
-            "@boringssl//:ssl",
+            "@ssl_lib//:crypto",
+            "@ssl_lib//:ssl",
         ],
         ":openssl": [
-            "@openssl//:crypto",
-            "@openssl//:ssl",
+            "@ssl_lib//:crypto",
+            "@ssl_lib//:ssl",
         ],
     }),
 )
diff --git a/MODULE.bazel b/MODULE.bazel
index 82fb1cc..dc64e2a 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -23,7 +23,6 @@ bazel_dep(name = "boost.pfr", version = "1.90.0.bcr.1")
 bazel_dep(name = "boost.system", version = "1.90.0.bcr.1")
 bazel_dep(name = "boost.throw_exception", version = "1.90.0.bcr.1")
 bazel_dep(name = "boost.variant2", version = "1.90.0.bcr.1")
-bazel_dep(name = "boringssl", version = "0.20251002.0")
-bazel_dep(name = "openssl", version = "3.5.5.bcr.0")
+bazel_dep(name = "ssl_lib", version = "1.0.0")
 bazel_dep(name = "platforms", version = "1.0.0")
 bazel_dep(name = "rules_cc", version = "0.2.16")
