diff --git a/src/cpp/security/artifact_providers/FileProvider.cpp b/src/cpp/security/artifact_providers/FileProvider.cpp
index e8f087f..450810d 100644
--- a/src/cpp/security/artifact_providers/FileProvider.cpp
+++ b/src/cpp/security/artifact_providers/FileProvider.cpp
@@ -77,7 +77,7 @@ X509* FileProvider::load_ca(
                             // Retrieve subject name for future use.
                             if (ca_sn.empty())
                             {
-                                X509_NAME* ca_subject_name = X509_get_subject_name(itmp->x509);
+                                const X509_NAME* ca_subject_name = X509_get_subject_name(itmp->x509);
                                 assert(ca_subject_name != nullptr);
                                 char* ca_subject_name_str = X509_NAME_oneline(ca_subject_name, 0, 0);
                                 assert(ca_subject_name_str != nullptr);
diff --git a/src/cpp/security/authentication/PKIDH.cpp b/src/cpp/security/authentication/PKIDH.cpp
index 6322c1e..fc2da7f 100644
--- a/src/cpp/security/authentication/PKIDH.cpp
+++ b/src/cpp/security/authentication/PKIDH.cpp
@@ -507,7 +507,7 @@ static bool adjust_participant_key(
 {
     assert(cert != nullptr);
 
-    X509_NAME* cert_sn = X509_get_subject_name(cert);
+    const X509_NAME* cert_sn = X509_get_subject_name(cert);
     assert(cert_sn != nullptr);
 
     unsigned char md[SHA256_DIGEST_LENGTH];
@@ -1203,7 +1203,7 @@ ValidationResult_t PKIDH::validate_local_identity(
         if ((*ih)->cert_ != nullptr)
         {
             // Get subject name.
-            X509_NAME* cert_sn = X509_get_subject_name((*ih)->cert_);
+            const X509_NAME* cert_sn = X509_get_subject_name((*ih)->cert_);
             assert(cert_sn != nullptr);
             char* cert_sn_str = X509_NAME_oneline(cert_sn, 0, 0);
             assert(cert_sn_str != nullptr);
@@ -1506,7 +1506,7 @@ ValidationResult_t PKIDH::validate_remote_identity(
         return ValidationResult_t::VALIDATION_FAILED;
     }
 
-    X509_NAME* cert_sn = X509_get_subject_name(rih->cert_);
+    const X509_NAME* cert_sn = X509_get_subject_name(rih->cert_);
     assert(cert_sn != nullptr);
     char* cert_sn_str = X509_NAME_oneline(cert_sn, 0, 0);
     assert(cert_sn_str != nullptr);
@@ -1931,7 +1931,7 @@ ValidationResult_t PKIDH::validate_remote_identity(
         return ValidationResult_t::VALIDATION_FAILED;
     }
 
-    X509_NAME* cert_sn = X509_get_subject_name(rih->cert_);
+    const X509_NAME* cert_sn = X509_get_subject_name(rih->cert_);
     assert(cert_sn != nullptr);
     char* cert_sn_str = X509_NAME_oneline(cert_sn, 0, 0);
     assert(cert_sn_str != nullptr);
