diff --git a/BUILD.bazel b/BUILD.bazel
index 2c8793c..735b498 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -105,7 +105,8 @@ cc_library(
         "@fastcdr",
         "@foonathan_memory",
         "@nlohmann_json//:singleheader-json",
-        "@openssl",
+        "@ssl_lib//:crypto",
+        "@ssl_lib//:ssl",
         "@sqlite3",
         "@tinyxml2",
     ],
diff --git a/MODULE.bazel b/MODULE.bazel
index e555210..5c27a9b 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -24,7 +24,7 @@ bazel_dep(name = "cmake_configure_file", version = "0.1.6")
 bazel_dep(name = "fastcdr", version = "2.3.5.bcr.0")
 bazel_dep(name = "foonathan_memory", version = "0.7.3.bcr.2")
 bazel_dep(name = "nlohmann_json", version = "3.12.0.bcr.1")
-bazel_dep(name = "openssl", version = "3.5.5.bcr.0")
+bazel_dep(name = "ssl_lib", version = "1.0.0")
 bazel_dep(name = "platforms", version = "1.0.0")
 bazel_dep(name = "rules_cc", version = "0.2.16")
 bazel_dep(name = "sqlite3", version = "3.51.2")
