diff --git a/BUILD.bazel b/BUILD.bazel
index c950109..2b3909c 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -125,28 +125,7 @@ genrule(
         ":proto_h",
     ],
     outs = ["rounding.h"],
-    cmd = """
-cp $(location :proto_h) proto.h
-CC="$${CC:-cc}"
-for r in 0 1 3; do
-  if "$$CC" -o rounding -DEXTRA_ROUNDING=$$r -DRSYNC_NO_OPENSSL -I. $(location rounding.c) >rounding.out 2>&1; then
-    echo "#define EXTRA_ROUNDING $$r" > $@
-    if test -n "$${HOME-}" && test -f "$${HOME-}/build_farm/build_test.fns"; then
-      echo "EXTRA_ROUNDING is $$r" >&2
-    fi
-    break
-  fi
-done
-rm -f rounding
-if test -f "$@"; then
-  :
-else
-  cat rounding.out >&2
-  echo "Failed to create rounding.h!" >&2
-  exit 1
-fi
-rm -f rounding.out proto.h
-""",
+    cmd = "echo '#define EXTRA_ROUNDING 1' > $@",
 )
 
 genrule(
@@ -230,9 +209,6 @@ cc_library(
         "popt",
         "zlib",
     ],
-    linkopts = [
-        "-liconv",
-    ],
     deps = [
         ":popt",
         ":zlib",
diff --git a/config.h b/config.h
index 51d811a..05f0b85 100644
--- a/config.h
+++ b/config.h
@@ -75,7 +75,7 @@
 #define HAVE_C99_VSNPRINTF 1
 
 /* Define to 1 if you have the 'chflags' function. */
-#define HAVE_CHFLAGS 1
+/* #undef HAVE_CHFLAGS */
 
 /* Define to 1 if you have the 'chmod' function. */
 #define HAVE_CHMOD 1
@@ -140,7 +140,7 @@
 #define HAVE_GETADDRINFO 1
 
 /* Define to 1 if you have the 'getattrlist' function. */
-#define HAVE_GETATTRLIST 1
+/* #undef HAVE_GETATTRLIST */
 
 /* Define to 1 if you have the 'getcwd' function. */
 #define HAVE_GETCWD 1
@@ -206,7 +206,7 @@
 #define HAVE_LANGINFO_H 1
 
 /* Define to 1 if you have the 'lchmod' function. */
-#define HAVE_LCHMOD 1
+/* #undef HAVE_LCHMOD */
 
 /* Define to 1 if you have the 'lchown' function. */
 #define HAVE_LCHOWN 1
@@ -215,7 +215,7 @@
 #define HAVE_LCHOWN 1
 
 /* Define to 1 if you have the <libcharset.h> header file. */
-#define HAVE_LIBCHARSET_H 1
+/* #undef HAVE_LIBCHARSET_H */
 
 /* Define to 1 if you have the <limits.h> header file. */
 #define HAVE_LIMITS_H 1
@@ -257,7 +257,7 @@
 /* #undef HAVE_LINUX_XATTRS */
 
 /* Define to 1 if you have the 'locale_charset' function. */
-#define HAVE_LOCALE_CHARSET 1
+/* #undef HAVE_LOCALE_CHARSET */
 
 /* Define to 1 if you have the <locale.h> header file. */
 #define HAVE_LOCALE_H 1
@@ -346,10 +346,10 @@
 #define HAVE_OPENSSL_MD5_H 1
 
 /* true if you have Mac OS X ACLs */
-#define HAVE_OSX_ACLS 1
+/* #undef HAVE_OSX_ACLS */
 
 /* True if you have Mac OS X xattrs */
-#define HAVE_OSX_XATTRS 1
+/* #undef HAVE_OSX_XATTRS */
 
 /* Define to 1 if the system has the type 'pid_t'. */
 #define HAVE_PID_T 1
@@ -379,7 +379,7 @@
 #define HAVE_SECURE_MKSTEMP 1
 
 /* Define to 1 if you have the 'setattrlist' function. */
-#define HAVE_SETATTRLIST 1
+/* #undef HAVE_SETATTRLIST */
 
 /* Define to 1 if you have the 'setenv' function. */
 #define HAVE_SETENV 1
@@ -394,7 +394,7 @@
 #define HAVE_SETLOCALE 1
 
 /* Define to 1 if you have the 'setmode' function. */
-#define HAVE_SETMODE 1
+/* #undef HAVE_SETMODE */
 
 /* Define to 1 if you have the 'setsid' function. */
 #define HAVE_SETSID 1
@@ -418,10 +418,10 @@
 #define HAVE_SNPRINTF 1
 
 /* Do we have sockaddr_in.sin_len? */
-#define HAVE_SOCKADDR_IN_LEN 1
+/* #undef HAVE_SOCKADDR_IN_LEN */
 
 /* Do we have sockaddr.sa_len? */
-#define HAVE_SOCKADDR_LEN 1
+/* #undef HAVE_SOCKADDR_LEN */
 
@@ -466,10 +466,10 @@
 #define HAVE_STRING_H 1
 
 /* Define to 1 if you have the 'strlcat' function. */
-#define HAVE_STRLCAT 1
+/* #undef HAVE_STRLCAT */
 
 /* Define to 1 if you have the 'strlcpy' function. */
-#define HAVE_STRLCPY 1
+/* #undef HAVE_STRLCPY */
 
 /* Define to 1 if you have the 'strpbrk' function. */
 #define HAVE_STRPBRK 1
@@ -490,10 +490,10 @@
 /* #undef HAVE_STRUCT_STAT_ST_MTIMENSEC */
 
 /* Define to 1 if 'st_mtimespec.tv_nsec' is a member of 'struct stat'. */
-#define HAVE_STRUCT_STAT_ST_MTIMESPEC_TV_NSEC 1
+/* #undef HAVE_STRUCT_STAT_ST_MTIMESPEC_TV_NSEC */
 
 /* Define to 1 if 'st_mtim.tv_nsec' is a member of 'struct stat'. */
-/* #undef HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC */
+#define HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC 1
 
 /* Define to 1 if 'st_rdev' is a member of 'struct stat'. */
 #define HAVE_STRUCT_STAT_ST_RDEV 1
@@ -502,10 +502,10 @@
 #define HAVE_STRUCT_UTIMBUF 1
 
 /* Define to 1 if you have the <sys/acl.h> header file. */
-#define HAVE_SYS_ACL_H 1
+/* #undef HAVE_SYS_ACL_H */
 
 /* Define to 1 if you have the <sys/attr.h> header file. */
-#define HAVE_SYS_ATTR_H 1
+/* #undef HAVE_SYS_ATTR_H */
 
 /* Define to 1 if you have the <sys/dir.h> header file, and it defines 'DIR'.
    */
@@ -524,7 +524,7 @@
 #define HAVE_SYS_FILE_H 1
 
 /* Define to 1 if you have the <sys/filio.h> header file. */
-#define HAVE_SYS_FILIO_H 1
+/* #undef HAVE_SYS_FILIO_H */
 
 /* Define to 1 if you have the <sys/ioctl.h> header file. */
 #define HAVE_SYS_IOCTL_H 1
@@ -655,7 +655,7 @@
 /* Define to 1 if `major', `minor', and `makedev' are declared in
    <sysmacros.h>. */
-/* #undef MAJOR_IN_SYSMACROS */
+#define MAJOR_IN_SYSMACROS 1
 
 /* Define to 1 if makedev() takes 3 args */
 /* #undef MAKEDEV_TAKES_3_ARGS */
@@ -778,13 +778,13 @@
 #define STDC_HEADERS 1
 
 /* Define to 1 to add support for ACLs */
-#define SUPPORT_ACLS 1
+/* #undef SUPPORT_ACLS */
 
 /* Undefine if you do not want LZ4 compression. By default this is defined. */
 #define SUPPORT_LZ4 1
 
 /* Define to 1 to add support for extended attributes */
-#define SUPPORT_XATTRS 1
+/* #undef SUPPORT_XATTRS */
 
 /* Undefine if you do not want xxhash checksums. By default this is defined.
    */
